src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -0,0 +1,408 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
```

Comment:
Nit: Doesn't seem to belong to Interop : https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md

> With few exceptions, the only methods that should be defined in these interop types are DllImports.
> * Exceptions are limited to times when most or every consumer of a particular DllImport will need to wrap its invocation in a helper, e.g. to provide additional marshaling support, to hide thread-safety issues in the underlying OS implementation, to do any required manipulation of safe handles, etc. In such cases, the DllImport should be private whenever possible rather than internal, with the helper code exposed to consumers rather than having the DllImport exposed directly.

For other System.Net implementations we've placed this kind of code in the PAL layer.